src/compiler/transformers/declarations.ts

```diff
@@ -0,0 +1,1276 @@
+                        enclosingDeclaration = input.trueType;
+                        const trueType = visitNode(input.trueType, visitDeclarationComponents);
+                        enclosingDeclaration = oldEnclosingDecl;
+                        const falseType = visitNode(input.falseType, visitDeclarationComponents);
+                        return cleanup(updateConditionalTypeNode(input, checkType, extendsType, trueType, falseType));
+                    }
+                    default: Debug.assertNever(input, `Attempted to process unhandled node kind: ${(ts as any).SyntaxKind[(input as any).kind]}`);
+                }
+            }
+
+            return cleanup(visitEachChild(input, visitDeclarationComponents, context));
+
+            function cleanup<T extends Node>(returnValue: T | undefined): T {
+                if (returnValue && canProdiceDiagnostic && hasDynamicName(input as Declaration)) {
+                    checkName(input as declarations.DeclarationDiagnosticProducing);
+                }
+                if (isEnclosingDeclaration(input)) {
+                    enclosingDeclaration = previousEnclosingDeclaration;
+                }
+                if (canProdiceDiagnostic && !suppressNewDiagnosticContexts) {
+                    getSymbolAccessibilityDiagnostic = oldDiag;
+                }
+                if (shouldEnterSuppressNewDiagnosticsContextContext) {
+                    suppressNewDiagnosticContexts = oldWithinObjectLiteralType;
+                }
+                if (returnValue === input) {
+                    return returnValue;
+                }
+                return returnValue && setOriginalNode(preserveJsDoc(returnValue, input), input);
+            }
+        }
+
+        function isPrivateMethodTypeParameter(node: TypeParameterDeclaration) {
+            return node.parent.kind === SyntaxKind.MethodDeclaration && hasModifier(node.parent, ModifierFlags.Private);
+        }
+
+        function updateModifiers<T extends Node>(updated: T, input: T): T {
+            const modifiers = ensureModifiers(input);
+            if (modifiers !== input.modifiers) {
+                if (updated === input) {
+                    updated = getMutableClone(input);
+                    updated.decorators = undefined;
+                }
+                updated.modifiers = createNodeArray(modifiers);
+            }
+            return updated;
+        }
+
+        function visitDeclarationStatements(input: Node): VisitResult<Node> {
+            if (!isPreservedDeclarationStatement(input)) {
+                // return undefined for unmatched kinds to omit them from the tree
+                return;
+            }
+            if (shouldStripInternal(input)) return;
+
+            switch (input.kind) {
+                case SyntaxKind.ExportDeclaration: {
+                    if (!resultHasExternalModuleIndicator && isSourceFile(input.parent)) {

```

Comment:
Technically I don't think you need to check `resultHasExternalModuleIndicator` before setting it to `true` but I guess it can save some work in the cases where you need to check modifiers.